### ISSUE TITLE
Azure: azure.storage-container policy not working

### ISSUE THREAD

**Issue by tomarv2** [cloud-custodian/cloud-custodian#5559]
Policy:

```
vars:
  notify: &notify
    type: notify
    subject: "[Azure Policy] [Subscription: {{ account_id }}] Storage policy"
    violation_desc: "Storage policy"
    slack_template: db_slack
    #template: db_email.html
    to:
      - varun@tomarv2.com
    transport:
      type: asq
      queue: https://tomarv2.queue.core.windows.net/notifications

policies:
  - name: custom-storage-container-public
    description: |
      Find all storage container with public access
    resource: azure.storage-container
    mode:
      type: container-event
      events: [{
        resourceProvider: 'Microsoft.Storage/storageAccounts',
        event: 'write'
        }]
    filters:
      - type: value
        key: properties.networkAcls.ipRules
        value_type: size
        op: equal
        value: 0
    actions:
      - <<: *notify
        subject: "[Azure Policy] [Subscription: {{ account_id }}] Container(s) with public access enabled"
        violation_desc: "Container(s) with public access enabled"
```

Error in ACI logs:
```
Policy Failed: custom-storage-container-public
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/container_host/host.py", line 405, in run_policy
    policy.push(event, context)
  File "/usr/local/lib/python3.7/site-packages/c7n/policy.py", line 1011, in push
    return mode.run(event, lambda_ctx)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/container_host/modes.py", line 96, in run
    return AzureModeCommon.run_for_event(self.policy, event)
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/policy.py", line 266, in run_for_event
    [AzureModeCommon.extract_resource_id(policy, event)])
  File "/usr/local/lib/python3.7/site-packages/c7n_azure/policy.py", line 259, in extract_resource_id
    return re.search(extract_regex, event['subject'], re.IGNORECASE).group()
AttributeError: 'NoneType' object has no attribute 'group'
```

**Comment by kapilt** [comment]
do you have actual event? available from the logs (sanitized)

ping @cloud-custodian/azure

**Comment by stefangordon** [comment]
Possibly something has changed with the event syntax.  You might be able to find of the events in the queue still since likely the error prevented it from dequeue.  Or activity logs.

**Comment by tomarv2** [comment]
@kapilt storage container does exist, as it is event-based so the policy was triggered when it found something relevant.

@stefangordon I can see public container getting created in activity logs.

**Comment by stefangordon** [comment]
If you are able to share the json from the activity log it would help, as the error is in parsing the event (remove any non-public data before sharing :) )

**Comment by tomarv2** [comment]
<img width="1630" alt="Screen Shot 2020-04-05 at 10 05 27 AM" src="https://user-images.githubusercontent.com/11890959/78505018-5dd5a800-7725-11ea-9ff9-8f4648d5460a.png">


Activity logs json
```
{
    "authorization": {
        "action": "Microsoft.Storage/storageAccounts/blobServices/containers/write",
        "scope": "/subscriptions/1234567890/resourceGroups/cloud-custodian/providers/Microsoft.Storage/storageAccounts/tomarv2c7nstorage/blobServices/default/containers/tesdt3"
    },
    "caller": "varun@tomarv2.com",
    "channels": "Operation",
    "claims": {
        "aud": "https://management.core.windows.net/",
        "iss": "https://sts.windows.net/9f37a3/",
        "iat": "1586098049",
        "nbf": "1586098049",
        "exp": "1586098949",
        "http://schemas.microsoft.com/claims/authnclassreference": "1",
        "aio": "ATQAy/8PAAAAWX",
        "http://schemas.microsoft.com/claims/authnmethodsreferences": "pwd",
        "appid": "c44b408",
        "appidacr": "2",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname": "Tomar",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname": "Varun",
        "ipaddr": "11.202.11.11",
        "name": "Varun Tomar",
        "http://schemas.microsoft.com/identity/claims/objectidentifier": "52626d33",
        "puid": "1003200",
        "http://schemas.microsoft.com/identity/claims/scope": "user_impersonation",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier": "iphGposnJ10q",
        "http://schemas.microsoft.com/identity/claims/tenantid": "9f37a392",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name": "varun@tomarv2.com",
        "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn": "varun@tomarv2.com",
        "uti": "HlYX",
        "ver": "1.0"
    },
    "correlationId": "1a91dd65",
    "description": "",
    "eventDataId": "3806bad",
    "eventName": {
        "value": "EndRequest",
        "localizedValue": "End request"
    },
    "category": {
        "value": "Administrative",
        "localizedValue": "Administrative"
    },
    "eventTimestamp": "2020-04-05T14:55:36.2549472Z",
    "id": "/subscriptions/1234567890/resourceGroups/cloud-custodian/providers/Microsoft.Storage/storageAccounts/tomarv2c7nstorage/blobServices/default/containers/tesdt3/events/3806bad7",
    "level": "Informational",
    "operationId": "1a91dd65",
    "operationName": {
        "value": "Microsoft.Storage/storageAccounts/blobServices/containers/write",
        "localizedValue": "Put blob container"
    },
    "resourceGroupName": "cloud-custodian",
    "resourceProviderName": {
        "value": "Microsoft.Storage",
        "localizedValue": "Microsoft.Storage"
    },
    "resourceType": {
        "value": "Microsoft.Storage/storageAccounts/blobServices/containers",
        "localizedValue": "Microsoft.Storage/storageAccounts/blobServices/containers"
    },
    "resourceId": "/subscriptions/1234567890/resourceGroups/cloud-custodian/providers/Microsoft.Storage/storageAccounts/tomarv2c7nstorage/blobServices/default/containers/tesdt3",
    "status": {
        "value": "Succeeded",
        "localizedValue": "Succeeded"
    },
    "subStatus": {
        "value": "Created",
        "localizedValue": "Created (HTTP Status Code: 201)"
    },
    "submissionTimestamp": "2020-04-05T14:56:28.1527156Z",
    "subscriptionId": "1234567890",
    "tenantId": "9f37a392",
    "properties": {
        "statusCode": "Created",
        "serviceRequestId": "9aa59c79"
    },
    "relatedEvents": []
}
```

**Comment by stefangordon** [comment]
Thanks! We'll try to get this reproduced in the next day or two.

**Comment by tomarv2** [comment]
@stefangordon did you get the chance to look at it..

**Comment by stefangordon** [comment]
tldr; I think you wanted the resource type `azure.storage` not `azure.storage-container`

Depending on what you are trying to do we may or may not support it yet.

You are watching for events on containers being created, but you are filtering on firewall values which are storage account configurations, not container configurations.

Storage accounts can be "public" by nature of their firewall configuration, whereas containers can be "public" by nature of their Access Policy.

On a container you can do this:

```
    filters:
      - type: value
        key: properties.publicAccess
        op: not-equal
        value: None   # Possible values: Blob, Container, None
```

And on a storage account you can check the firewall as you were doing.

In your example policy you are catching events for **Storage Accounts** but your resource type is set to **Storage Container**.

A correct **Storage Container** event mode would be as follows, but we have a bug preventing this from working still #4617 
```
    mode:
      type: container-event
      events: [{
        resourceProvider: 'Microsoft.Storage/storageAccounts/blobServices/containers',
        event: 'write'
        }]
```

**Comment by tomarv2** [comment]
thanks @stefangordon scheduled based policy worked.

**Comment by stefangordon** [comment]
Great, I also started adding container event support in, you can follow https://github.com/cloud-custodian/cloud-custodian/pull/5610
